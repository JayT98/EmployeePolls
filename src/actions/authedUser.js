export const LOGOUT_AUTHED_USER = "LOGOUT_AUTHED_USER";
export const SET_AUTHED_USER = "SET_AUTHED_USER";

export function setAuthedUser(authedUser){
    return {
        type: SET_AUTHED_USER,
        authedUser
    }
}

export function logoutAuthedUser() {
    return {
        type: LOGOUT_AUTHED_USER
    }
}

export function handleLogin(username, password){
    return(dispatch, getState) => {
        const {users} = getState();

        const user = Object.values(users).find((user) =>
            user.id === username && user.password === password
        );

        if(!!user){
            return dispatch(setAuthedUser(user));
        }
    };
}

export function handleLogout() {
    return (dispatch) => {
        return dispatch(logoutAuthedUser());
    }
}
