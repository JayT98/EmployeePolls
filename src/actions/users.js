export const ADD_QUESTION_USER = "ADD_QUESTION_USER";
export const ADD_ANSWER_USER = "ADD_ANSWER_USER";
export const RECEIVE_USERS = "RECEIVE_USERS";

export function receiveUsers(users) {
    return {
        type: RECEIVE_USERS,
        users
    }
}

export function addQuestionUser({author, id}) {
    return {
        type: ADD_QUESTION_USER,
        author,
        qid: id,
    };
}

export function addAnswerUser(authedUser, qid, answer) {
    return {
        type: ADD_ANSWER_USER,
        authedUser,
        qid,
        answer
    };
}
