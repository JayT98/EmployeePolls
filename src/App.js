import React, {useEffect} from 'react';
import {Route, Routes} from 'react-router-dom';
import Nav from './components/Nav';
import Dashboard from './components/Dashboard';
import Leaderboard from './components/Leaderboard';
import Login from './components/Login';
import Error404 from './components/404';
import PollPage from './components/PollPage';
import NewPoll from './components/NewPoll';
import PrivateRoute from './components/PrivateRoute';
import {connect} from 'react-redux';
import { handleInitialData } from './actions/shared';


function App({dispatch, loggedIn}) {
  useEffect(() => {
    dispatch(handleInitialData());
  });
  
  return (
    <div className='container mx-auto py-4'>
      {loggedIn && <Nav/>}
      <Routes>
        <Route path='/login' exact element={<Login/>}/>
        <Route path='/' element={<PrivateRoute><Dashboard/></PrivateRoute>}/>
        <Route path='/leaderboard' exact element={<PrivateRoute><Leaderboard/></PrivateRoute>}/>
        <Route path='/questions/:id' element={<PrivateRoute><PollPage/></PrivateRoute>}/>
        <Route path="/new" exact element={<PrivateRoute><NewPoll/></PrivateRoute>}/>
        <Route path="/404" exact element={<Error404/>}/>
      </Routes>
    </div>
  );
}

const mapStateToProps = ({authedUser}) => ({
  loggedIn: !!authedUser,
});

export default connect(mapStateToProps)(App);
