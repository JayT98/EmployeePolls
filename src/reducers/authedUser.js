import {LOGOUT_AUTHED_USER, SET_AUTHED_USER} from '../actions/authedUser';

export default function authedUser(state = null, action) {
    switch (action.type) {
        case SET_AUTHED_USER:
            return action.authedUser;
        case LOGOUT_AUTHED_USER:
            return null;
        default:
            return state;
    }
}