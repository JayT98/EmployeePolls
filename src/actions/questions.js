import {saveQuestion, saveQuestionAnswer} from '../data/api';
import {addAnswerUser, addQuestionUser} from './users';

export const RECEIVE_QUESTIONS = 'RECEIVE_QUESTIONS';
export const ADD_QUESTION = 'ADD_QUESTION';
export const ADD_ANSWER_QUESTION = 'ADD_ANSWER_QUESTION';

export function receiveQuestions(questions) {
    return {
        type: RECEIVE_QUESTIONS,
        questions
    };
}

function addQuestion(question){
    return {
        type: ADD_QUESTION,
        question
    };
}

export function addAnswerQuestion(author, qid, answer) {
    return {
        type: ADD_ANSWER_QUESTION,
        author,
        qid,
        answer
    };
}

export function handleAddQuestion(firstOption, secondOption) {
    return (dispatch, getState) => {
        const {authedUser} = getState();

        return saveQuestion(firstOption, secondOption, authedUser)
                .then((question) => {
                    dispatch(addQuestion(question));
                    dispatch(addQuestionUser(question))
                });
    };
}

export function handleAddAnswer(questionId, answer) {
    return (dispatch, getState) => {
        const {authedUser} = getState();

        return saveQuestionAnswer(authedUser.id, questionId, answer)
                .then(() => {
                    dispatch(addAnswerQuestion(authedUser.id, questionId, answer));
                    dispatch(addAnswerUser(authedUser.id, questionId, answer))
                });
    };
}