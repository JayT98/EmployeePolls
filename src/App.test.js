import { render, screen } from '@testing-library/react';
import App from './App';
import React from 'react';
import {Provider} from "react-redux";
import {store} from "./store";
import {BrowserRouter} from "react-router-dom";
import {setAuthedUser} from "./actions/authedUser";

describe("App", () => {
  it("render the component", () => {
      const component = render(
          <Provider store={store}>
              <BrowserRouter>
                  <App/>
              </BrowserRouter>
          </Provider>
      );
      expect(component).toBeDefined();
      expect(component).toMatchSnapshot();
  });

  it("show Login page when not logged in", () => {
      const component = render(
          <Provider store={store}>
              <BrowserRouter>
                  <App/>
              </BrowserRouter>
          </Provider>
      );
      const heading = component.getByTestId("login-heading");
      expect(heading).toBeInTheDocument();
  });

  it("should show Dashboard page when logged in", () => {
      store.dispatch(setAuthedUser({id: "", password: ""}));

      const component = render(
          <Provider store={store}>
              <BrowserRouter>
                  <App/>
              </BrowserRouter>
          </Provider>
      );

      const heading = component.getByTestId("heading");
      expect(heading).toBeInTheDocument();
  });
});