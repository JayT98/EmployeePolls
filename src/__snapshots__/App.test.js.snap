// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto py-4"
      >
        <div>
          <h1
            class="text-3xl font-bold mt-9 text-center"
          >
            Employee Polls
          </h1>
          <div
            class="w-full h-max mt-10 flex justify-center"
          >
            <img
              alt="Avatar_brand"
              src="avatar_brand.png"
            />
          </div>
          <h2
            class="text-3xl font-bold mt-9 mb-5 text-center"
            data-testid="login-heading"
          >
            Log In
          </h2>
          <form>
            <div>
              <label
                class="block mb-5 text-base font-bold text-black text-center"
                for="username"
              >
                User
              </label>
              <div
                class="mt-1"
              >
                <input
                  class="px-4 py-2 bg-white  border shardow-sm border-slate-400 placeholder-slate-300 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-green-400 focus:ring-green-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-red-700 invalid:text-red-600 focus:invalid:border-red-500 focus:invalid:ring-red-500 disabled:shadow-none "
                  data-testid="username"
                  id="username"
                  name="username"
                  type="text"
                  value="sarahedo"
                />
              </div>
            </div>
            <div>
              <label
                class="block mb-5 mt-5 text-base font-bold text-black text-center"
                for="password"
              >
                Password
              </label>
              <div
                class="mt-1"
              >
                <input
                  class="px-4 py-2 bg-white  border shardow-sm border-slate-400 placeholder-slate-300 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-green-400 focus:ring-green-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-red-700 invalid:text-red-600 focus:invalid:border-red-500 focus:invalid:ring-red-500 disabled:shadow-none "
                  data-testid="password"
                  id="password"
                  name="password"
                  type="password"
                  value="password123"
                />
              </div>
            </div>
            <div
              class="mt-6 text-center"
            >
              <button
                class="bg-green-500 hover:bg-green-700 px-7 py-3 text-sm leading-5 font-bold rounded-md text-white"
                data-testid="submit"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto py-4"
    >
      <div>
        <h1
          class="text-3xl font-bold mt-9 text-center"
        >
          Employee Polls
        </h1>
        <div
          class="w-full h-max mt-10 flex justify-center"
        >
          <img
            alt="Avatar_brand"
            src="avatar_brand.png"
          />
        </div>
        <h2
          class="text-3xl font-bold mt-9 mb-5 text-center"
          data-testid="login-heading"
        >
          Log In
        </h2>
        <form>
          <div>
            <label
              class="block mb-5 text-base font-bold text-black text-center"
              for="username"
            >
              User
            </label>
            <div
              class="mt-1"
            >
              <input
                class="px-4 py-2 bg-white  border shardow-sm border-slate-400 placeholder-slate-300 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-green-400 focus:ring-green-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-red-700 invalid:text-red-600 focus:invalid:border-red-500 focus:invalid:ring-red-500 disabled:shadow-none "
                data-testid="username"
                id="username"
                name="username"
                type="text"
                value="sarahedo"
              />
            </div>
          </div>
          <div>
            <label
              class="block mb-5 mt-5 text-base font-bold text-black text-center"
              for="password"
            >
              Password
            </label>
            <div
              class="mt-1"
            >
              <input
                class="px-4 py-2 bg-white  border shardow-sm border-slate-400 placeholder-slate-300 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-green-400 focus:ring-green-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-red-700 invalid:text-red-600 focus:invalid:border-red-500 focus:invalid:ring-red-500 disabled:shadow-none "
                data-testid="password"
                id="password"
                name="password"
                type="password"
                value="password123"
              />
            </div>
          </div>
          <div
            class="mt-6 text-center"
          >
            <button
              class="bg-green-500 hover:bg-green-700 px-7 py-3 text-sm leading-5 font-bold rounded-md text-white"
              data-testid="submit"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
